packages/remix/src/useForm.tsx

```diff
@@ -107,6 +107,10 @@ export function useForm<FormInputData extends FieldValues, FormOutputData>(
         if (!formData)
           throw new Error("Missing form data. This is likely a bug in RVF");
         if (rvfOpts.id) formData.set(FORM_ID_FIELD_NAME, rvfOpts.id);
+
+        if (rvfOpts.encType === "application/json") {
+          return Object.fromEntries(formData.entries()) as GenericObject;
```

Comment:
I think this is the right place to change, but this won't handle nested data. You should be able to use `preprocessFormData` from `@rvf/core` instead.

Reply:
yep, done